www/c-icap/src/opnsense/service/templates/OPNsense/CICAP/virus_scan.conf

```diff
@@ -0,0 +1,25 @@
+{% if helpers.exists('OPNsense.cicap.antivirus.enable_clamav') and OPNsense.cicap.antivirus.enable_clamav == '1' %}
+Service antivirus_module virus_scan.so
+ServiceAlias srv_clamav virus_scan
+ServiceAlias  avscan virus_scan?allow204=on&sizelimit=off&mode=simple
+virus_scan.ScanFileTypes TEXT DATA EXECUTABLE ARCHIVE GIF JPEG MSOFFICE
+
+{% if helpers.exists('OPNsense.cicap.antivirus.sendpercentdata') and OPNsense.cicap.antivirus.sendpercentdata != '' %}
+virus_scan.SendPercentData 5
+{% endif %}
+{% if helpers.exists('OPNsense.cicap.antivirus.startsendpercentdataafter') and OPNsense.cicap.antivirus.startsendpercentdataafter != '' %}
+virus_scan.StartSendPercentDataAfter 2M
+{% endif %}
+{% if helpers.exists('OPNsense.cicap.antivirus.allow204responses') and OPNsense.cicap.antivirus.allow204responses != '' %}
+virus_scan.Allow204Responces on
+{% endif %}
+{% if helpers.exists('OPNsense.cicap.antivirus.passonerror') and OPNsense.cicap.antivirus.passonerror != '' %}
+virus_scan.PassOnError off
+{% endif %}
+{% if helpers.exists('OPNsense.cicap.antivirus.cmaxobjectsize') and OPNsense.cicap.antivirus.maxobjectsize != '' %}
+virus_scan.MaxObjectSize  5M
+{% endif %}
+Module common clamd_mod.so
+clamd_mod.ClamdHost 127.0.0.1
+clamd_mod.ClamdPort 3310
```

Comment:
We are trying to avoid too much crosstalk. The default should connect to the local clamav as the plugin does it by default, but this when configured shall point somewhere else, worst case a second box where the plugin is running. In any case both plugins should be put into a working state before we define their interaction. We chose separate plugins for flexibility. Ease of use must come later. 🙂